### PR TITLE
Feat(Tekla): Add support for Tekla v2022

### PR DESCRIPTION
**Pull request by connorivy**
## Changes:

- add connector and converter v2022 for tekla structures
- delete unused items and consolidate repeated code
- change post build events to accomodate multiple tekla install locations

## Validation of changes:

able to open and receive/send to tekla structures 2022

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

## References

This task was started here https://github.com/specklesystems/speckle-sharp/pull/1979 by Rey but wasn't totally finished

### CODE REVIEW

File: ConnectorTeklaStructures/ConnectorTeklaStructures2022/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+// The following GUID is for the ID of the typelib if this project is exposed to COM
+[assembly: Guid("3af1ef30-0906-4926-a02c-4e3ad666352a")]
```
Comment (connorivy) [inline ConnectorTeklaStructures/ConnectorTeklaStructures2022/Properties/AssemblyInfo.cs:23]: I maybe need an explaination of what this is for. Currently, this guid is the same as the connector2021 assembly guid. Idk if that matters

File: ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj [hunk 2]
```diff
@@ -0,0 +1,119 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\packages\TeklaOpenAPI.2021.0.1\build\TeklaOpenAPI.props" Condition="Exists('..\packages\TeklaOpenAPI.2021.0.1\build\TeklaOpenAPI.props')" />
```
Comment (teocomi) [inline ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj:3]: Might need to be removed

File: ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    <PackageReference Include="Avalonia.Desktop">
+      <Version>0.10.18</Version>
+    </PackageReference>
+    <PackageReference Include="Avalonia.ReactiveUI">
+      <Version>0.10.18</Version>
+    </PackageReference>
+    <PackageReference Include="MSBuild.AssemblyVersion">
+      <Version>1.3.0</Version>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
```
Comment (teocomi) [inline ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj:86]: If we're removing the assemblyinfo.cs we can remove this

File: ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    </PackageReference>
+    <PackageReference Include="Stylet">
```
Comment (teocomi) [inline ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj:88]: can be removed

File: Objects/Converters/ConverterTeklaStructures/ConverterTeklaStructures2022/ConverterTeklaStructures2022.csproj [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  <PropertyGroup>
+    <IsDesktopBuild Condition="'$(IsDesktopBuild)' == ''">true</IsDesktopBuild>
+  </PropertyGroup>
+
+  <Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition="'$(IsDesktopBuild)' == true">
+    <Exec Condition="$([MSBuild]::IsOsPlatform('Windows'))" Command="xcopy /Y /S &quot;$(TargetDir)$(AssemblyName).dll&quot; &quot;$(AppData)\Speckle\Kits\Objects\&quot;&#xD;&#xA;xcopy /Y /S &quot;$(TargetDir)$(AssemblyName).dll&quot; &quot;$(ALLUSERSPROFILE)\Speckle\Kits\Objects\&quot;&#xD;&#xA;xcopy /Y /S &quot;$(TargetDir)\Objects.dll&quot; &quot;$(ALLUSERSPROFILE)\Speckle\Kits\Objects\&quot;" />
+  </Target>
+
+  <ItemGroup>
+    <PackageReference Include="Tekla.Structures" Version="2021.0.0" />
```
Comment (teocomi) [inline Objects/Converters/ConverterTeklaStructures/ConverterTeklaStructures2022/ConverterTeklaStructures2022.csproj:26]: Might need to target '22 nugets

**Review by teocomi — CHANGES_REQUESTED** [review]

File: Objects/Converters/ConverterTeklaStructures/ConverterTeklaStructures2022/ConverterTeklaStructures2022.csproj (hunk 5 above)
Comment (connorivy) [reply Objects/Converters/ConverterTeklaStructures/ConverterTeklaStructures2022/ConverterTeklaStructures2022.csproj]: done

File: ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj (hunk 4 above)
Comment (connorivy) [reply ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj]: done

File: ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj (hunk 3 above)
Comment (connorivy) [reply ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj]: done

File: ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj (hunk 2 above)
Comment (connorivy) [reply ConnectorTeklaStructures/ConnectorTeklaStructures2022/ConnectorTeklaStructures2022.csproj]: done

**Review by teocomi — APPROVED** [review]

**Pull request merged**